app/src/main/java/ch/sdp/vibester/activity/BuzzerScreenActivity.kt

```diff
@@ -144,13 +145,15 @@ class BuzzerScreenActivity : AppCompatActivity() {
     private fun setAnswerButton(answer: LinearLayout, button: Button, updater: BuzzerScoreUpdater, map: Map<Int, Int>) {
         button.setOnClickListener {
             answer.visibility = android.view.View.INVISIBLE
-            if (button.id==R.id.buttonCorrect) {
-                if (pressedBuzzer >= 0) {
-                    updater.updateScoresArray(pressedBuzzer)
-                    val view = map[pressedBuzzer]?.let { it1 -> findViewById<TextView>(it1) }
-                    if (view != null && updater.getMap().keys.contains(pressedBuzzer)) {view.text=updater.getMap()[pressedBuzzer].toString()}
-                }
+            if (pressedBuzzer >= 0) {
+                if(button.id==R.id.buttonCorrect)  {
+                    updater.updateScoresArray(pressedBuzzer, 1)
+                    winnerId = updater.getWinnerId()
+                } else {updater.updateScoresArray(pressedBuzzer, -1)}
+                val view = map[pressedBuzzer]?.let { it1 -> findViewById<TextView>(it1) }
+                if (view != null && updater.getMap().keys.contains(pressedBuzzer)) {view.text=updater.getMap()[pressedBuzzer].toString()}
             }

```

Comment:
Is this piece of code hard to test? You assign listener in onCreate, so if you do button.preform.click(), this code should be testable, no?